src/cli/commands/list.js

```diff
@@ -180,21 +181,33 @@ export function filterTree(tree: Tree, filters: Array<string>): boolean {
   return notDim && (found || hasChildren);
 }
 
+const getDevDeps = (manifest): Array<string> => Object.keys(manifest.devDependencies).map(key => `${key}@${manifest.devDependencies[key]}`);
+
 export async function run(config: Config, reporter: Reporter, flags: Object, args: Array<string>): Promise<void> {
   const lockfile = await Lockfile.fromDirectory(config.lockfileFolder, reporter);
   const install = new Install(flags, config, reporter, lockfile);
-  const {requests: depRequests, patterns, workspaceLayout} = await install.fetchRequestFromCwd();
+
+
+  const {requests: depRequests, patterns, manifest, workspaceLayout} = await install.fetchRequestFromCwd();
   await install.resolver.init(depRequests, {
     isFlat: install.flags.flat,
     isFrozen: install.flags.frozenLockfile,
     workspaceLayout,
   });
 
+  let pkgPatterns = [];
+  if (isProduction(process.env)) {
+    const devDeps = getDevDeps(manifest);
+    pkgPatterns = patterns.filter(pattern => !devDeps.includes(pattern));
```

Comment:
Good point. This is probably O(n)^n since `filter` loops over the entire collection and then `includes` checks the entire collection for existence there, too. 

`devDeps` being a set could get the filtering down to constant access to see if a package is a dependency, but I don't see how converting `patterns` to a set would benefit this particular piece of code, since its creation would be linear as well. I could be blindly missing something, though :)